docs/DeveloperGuide.adoc

```diff
@@ -192,6 +196,54 @@ image::add-ad.PNG[]
 ** Cons: Not as user-friendly.
 // end ::add[]
 
+// tag:stats[]
+=== Statistics feature : `stats`
+==== Implementation
+
+The stats feature allows the user to generate statistics about the eateries they have visited and reviews they have made at those visited eateries.
+
+Given below is an example usage scenario and how the stats mechanism behaves at each step.
+
+. The user launches the application for the first time. The `EateryList` will be initialised with the initial or previous json data stored.
+
+. The user executes `stats` to generate statistics for the current application state. The `EatMeParser` would return a `StatsCommand` to execute the command.
+
+. `StatsCommand` then creates `Statistics` to generate the different types of statistics needed. To create the different types of statistics, `Statistics` would, first, sort the eateries in `EateryList` based on different conditions i.e. categories, total expense, total number of visits. After sorting the eateries, `Statistics` would then use these sorted lists to generate the appropriate statistics.
+
+[NOTE]
+If the user executes the `stats` command in the Todo mode, `StatsCommand` would throw a `CommandException` to the user specifying that the user is in the wrong mode. In addition, if there is no eateries with any reviews in the application at the time of `stats` execution, the `StatsCommand` would throw a `CommandException` error saying that there is no data to generate statistics. Lastly, if there is any error during the generating of statistics, the `StatsCommand` will throw another `CommandException` saying that there is an unknown error and the application is unable to generate statistics.
```

Comment:
this one better ask daryl, idk if professional to put it under any error

Reply:
pinging @dvrylc